src/state/slices/assetsSlice/utils.ts

```diff
@@ -100,3 +103,36 @@ export const getFeeAssetByAssetId: GetFeeAssetByAssetId = (assetsById, assetId)
   })
   return assetsById[feeAssetId]
 }
+
+export const makeNftAssetsFromTxs = (txs: Transaction[]): AssetsState => {
+  return txs.reduce<AssetsState>(
+    (state, tx) => {
+      if (fromChainId(tx.chainId).chainNamespace !== CHAIN_NAMESPACE.Evm) return state
+
+      tx.transfers.forEach(transfer => {
+        if (state.byId[transfer.assetId] || !isNft(transfer.assetId)) return
```

Comment:
As discussed yesterday, this is perfectly fine IMO and mitigates the complexity of calling this with the full Tx history multiple times, sure this is still technically O(n) but this is akin to n getting smaller as this runs